spec/features/openid_connect/openid_connect_spec.rb

```diff
@@ -92,6 +92,34 @@ class MockSession; end
       expect(page.get_rack_session.keys).to include('sp')
     end
 
+    it 'redirects back to sp (not /account) if the user is already authenticated' do
+      client_id = 'urn:gov:gsa:openidconnect:sp:server'
+      user = user_with_2fa
+
+      IdentityLinker.new(user, client_id).link_identity
+      user.identities.last.update!(verified_attributes: ['email'])
+
+      visit openid_connect_authorize_path(
+        client_id: client_id,
+        response_type: 'code',
+        acr_values: Saml::Idp::Constants::LOA1_AUTHN_CONTEXT_CLASSREF,
+        scope: 'openid email',
+        redirect_uri: 'http://localhost:7654/auth/result',
+        state: SecureRandom.hex,
+        nonce: SecureRandom.hex,
+        prompt: 'select_account'
+      )
+
+      allow(FeatureManagement).to receive(:prefill_otp_codes?).and_return(true)
+      sign_in_user(user)
+
+      click_submit_default
+      expect(current_url).to start_with('http://localhost:7654/auth/result')
+
+      visit login_two_factor_path(otp_delivery_preference: 'sms')
+      expect(current_url).to start_with('http://localhost:7654/auth/result')
+    end
+
```

Comment:
Great catch! Could we please test both OIDC and SAML using the shared examples pattern we've been using? I would add a new shared example in `spec/support/shared_examples/sign_in.rb` called something like `submitting 2FA form when already fully authenticated` and then create a new file in `spec/features/two_factor_authentication/multiple_tabs_spec.rb` that would have something like this:
```ruby
require 'rails_helper'

feature 'user interacts with 2FA across multiple browser tabs' do
  it_behaves_like 'submitting 2FA form when already fully authenticated', :oidc
  it_behaves_like 'submitting 2FA form when already fully authenticated', :saml
end
```

Reply:
In the shared examples, you can take advantage of helpers like `visit_idp_from_sp_with_loa1(sp)`

Reply:
Great ideas.  I love the new structure.  To be clear though the redirects to /account were happening everywhere after sign in but before authentication on gets and posts for example: 
 just clicking a link (2fa options list).

Reply:
I reused whatever sign in method (personal key) was in the sign_in spec.

Reply:
Shall I leave the open_id_connect_spec in there or would you prefer I removed it?

Reply:
Yep, they will happen anywhere the `TwoFactorAuthenticatable` concern is included. The new tests look great. Yes, please remove this one.